### PR TITLE
[5.3]: fix a typo and make `name` optional in `gravity resource rm`

### PR DESCRIPTION
**Pull request by a-palchikov**
* Handle alternative resource names in modules interface.
 * Make 'name' flag optional for 'gravity resource rm'
 * Fix typo in a message banner

Complimentary PR: https://github.com/gravitational/gravity.e/pull/3903

### CODE REVIEW

File: lib/modules/modules.go [hunk 1]
```diff
@@ -39,8 +39,11 @@ type Modules interface {
 	DefaultAuthPreference(processMode string) (teleservices.AuthPreference, error)
 	// SupportedResources returns a list of resources that can be created/viewed
 	SupportedResources() []string
-	// SupportedResourcesToRemoves returns a list of resources that can be removed
+	// SupportedResourcesToRemove returns a list of resources that can be removed
 	SupportedResourcesToRemove() []string
+	// CanonicalKind translates the specified kind r to canonical form.
```
Comment (r0mant) [inline lib/modules/modules.go]: Remove "r" (from "the specified kind r").

File: lib/modules/modules.go [hunk 2]
```diff
@@ -108,6 +111,12 @@ func (m *defaultModules) SupportedConnectors() []string {
 	}
 }
 
+// CanonicalKind translates the specified kind r to canonical form.
```
Comment (r0mant) [inline lib/modules/modules.go]: Remove "r".

File: tool/gravity/cli/resources.go [hunk 3]
```diff
@@ -100,7 +101,7 @@ func CreateResource(
 
 // RemoveResource deletes resource by name
 func RemoveResource(env *localenv.LocalEnvironment, factory LocalEnvironmentFactory, kind string, name string, force bool, user string, manual, confirmed bool) error {
-	if kind == storage.KindRuntimeEnvironment {
+	if kind := modules.Get().CanonicalKind(kind); kind == storage.KindRuntimeEnvironment {
```
Comment (r0mant) [inline tool/gravity/cli/resources.go]: You could just do:

```go
if modules.Get().CanonicalKind(kind) == storage.KindRuntimeEnvironment {
```

b/c the kind variable is not in this block.

File: lib/ops/resources/resources.go [hunk 4]
```diff
@@ -76,6 +87,11 @@ func (r ListRequest) Check() error {
 	if r.Kind == "" {
 		return trace.BadParameter("resource kind is mandatory")
 	}
+	kind := modules.Get().CanonicalKind(r.Kind)
+	resources := modules.Get().SupportedResourcesToRemove()
```
Comment (r0mant) [inline lib/ops/resources/resources.go]: Why does "list" request check supported resources to "remove"?

Comment (a-palchikov) [reply lib/ops/resources/resources.go]: Thanks - I already spotted it.

**Pull request merged**